src/main/java/org/learn/example/connection/Person.java

```diff
@@ -0,0 +1,113 @@
+package org.learn.example.connection;
+
+import java.util.EnumSet;
+
+class Person{
+    private String firstName;
+    private String lastName;
+    private int age;
+    private double heightM;
+    private double weightKg;
+    private Wallet wallet;
+    private String denomination = Denomination.highestFirst.toString();
+
+    public Person(String firstName, String lastName, int age, double heightM, double weightKg, int initialMoney){
+        this.firstName = firstName;
+        this.lastName = lastName;
+        this.age = age;
+        this.heightM = heightM;
+        this.weightKg = weightKg;
+        if(initialMoney > 0){
+            this.wallet = new Wallet(initialMoney, this.denomination);
+        } else this.wallet = null;
+    }
+
+    enum Denomination {
+        highestFirst,
+        dollars,
+        twenties
+    }
+
+    public void setDenominationPreference(String denomination) {
+        if(EnumSet.allOf(Denomination.class).contains(Denomination.valueOf(denomination))) {
+            this.denomination = denomination;
+        } else {
+            throw new IllegalArgumentException(denomination + "は存在しません");
+        }
+    }
+
+    public int getCash(){
+        if(this.wallet == null) return 0;
+        return this.wallet.getTotalMoney();
+    }
+
+    public int[] spendMoney(int money) {
+        if(this.wallet == null) return null;
+        switch (this.denomination) {
```

Comment:
This is `Denomination` enum and logic is great for practice, but in actual implementation in might be even better to modularize it further and separate that logic from the person logic.